{
  try {
    FutureCallback<Void> callback=new FutureCallback<>();
    _endPoint.write(null,callback,bytes);
    callback.get();
  }
 catch (  InterruptedException x) {
    throw (IOException)new InterruptedIOException().initCause(x);
  }
catch (  ExecutionException x) {
    Throwable cause=x.getCause();
    if (cause instanceof IOException)     throw (IOException)cause;
 else     if (cause instanceof Exception)     throw new IOException(cause);
 else     throw (Error)cause;
  }
}
